napi.h

```diff
@@ -441,12 +494,105 @@ namespace Napi {
     void EnsureInfo() const;
   };
 
-  /*
-   * Holds a counted reference to a value; initially a weak reference unless otherwise specified.
-   * May be changed to/from a strong reference by adjusting the refcount. The referenced value
-   * is not immediately destroyed when the reference count is zero; it merely then eligible for
-   * GC if there are no other references to the value.
-   */
+  /// Wraps a JavaScript error object in a way that enables it to traverse a C++ stack and be
+  /// thrown and caught as a C++ exception.
+  ///
+  /// If a N-API call fails without executing any JavaScript code (for example due to an invalid
+  /// argument), then the N-API wrapper automatically converts and throws the error as a C++
+  /// exception of type `Napi::Error`. Or if a JavaScript function called by C++ code via N-API
+  /// throws a JavaScript exception, then the N-API wrapper automatically converts and throws it as
+  /// a C++ exception of type `Napi::Error`.
+  ///
+  /// If a C++ exception of type `Napi::Error` escapes from a N-API C++ callback, then the N-API
+  /// wrapper automatically converts and throws it as a JavaScript exception.
```

Comment:
By the way, should we catch all exceptions coming from C++ callbacks and convert them to JS errors? I don’t think there’s a way to do anything *more* meaningful (or at least I would assume JS engines don’t like it if the call stack unwinds outside of their control)